### PR TITLE
Refactor graphql queries

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-16T09:46:09Z" title="Friday, August 16th 2019, 11:46:09 am +02:00">Aug 16, 2019</time>_
_Merged <time datetime="2019-08-19T07:58:39Z" title="Monday, August 19th 2019, 9:58:39 am +02:00">Aug 19, 2019</time>_
---

- Remove duplicate queries
- Use smart query in pages/post/_id/_slug/index.vue to avoid multiple db requests for a post and its comments.
We cannot update the apollo cache with asyncData and smart queries have a prefetch policy set to true by default, which
means that they will resolve in a similar timeframe. https://stackoverflow.com/questions/55885337/in-nuxt-should-i-use-asyncdata-or-default-apollo-queries
https://vue-apollo.netlify.com/api/smart-query.html#options
https://vue-apollo.netlify.com/guide/ssr.html#vue-cli-plugin

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

File: webapp/components/comments/CommentList/index.vue [hunk 1]
```diff
@@ -41,29 +40,5 @@ export default {
...
-        return PostCommentsQuery(this.$i18n)
-      },
-      variables() {
-        return {
-          slug: this.post.slug,
-        }
-      },
-      fetchPolicy: 'cache-and-network',
-    },
-  },
```
Comment (Tirokk) [inline webapp/components/comments/CommentList/index.vue:67]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-16T10:49:49Z" title="Friday, August 16th 2019, 12:49:49 pm +02:00">Aug 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -41,29 +40,5 @@ export default {
   props: {
     post: { type: Object, default: () => {} },
   },
-  data() {
-    return {
-      comments: [],
-    }
-  },
-  watch: {
-    Post(post) {
-      const [first] = post
-      this.comments = (first && first.comments) || []
-    },
-  },
-  apollo: {
-    Post: {
-      query() {
-        return PostCommentsQuery(this.$i18n)
-      },
-      variables() {
-        return {
-          slug: this.post.slug,
-        }
-      },
-      fetchPolicy: 'cache-and-network',
-    },
-  },
```


Yesss, less code, less state, less bugs!

***State is the root of all evil***
root of all evil

File: webapp/pages/post/_id/_slug/index.vue [hunk 2]
```diff
@@ -122,29 +122,6 @@ export default {
...
-    const client = apolloProvider.defaultClient
-    const query = PostQuery($i18n)
-    const variables = { slug: params.slug }
-    const {
-      data: { Post },
-    } = await client.query({ query, variables })
-    if (Post.length <= 0) {
-      // TODO: custom 404 error page with translations
-      const message = 'This post could not be found'
-      return error({ statusCode: 404, message })
```
Comment (Tirokk) [inline webapp/pages/post/_id/_slug/index.vue:140]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-16T10:50:40Z" title="Friday, August 16th 2019, 12:50:40 pm +02:00">Aug 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -122,29 +122,6 @@ export default {
       this.title = this.post.title
     },
   },
-  async asyncData(context) {
-    const {
-      params,
-      error,
-      app: { apolloProvider, $i18n },
-    } = context
-    const client = apolloProvider.defaultClient
-    const query = PostQuery($i18n)
-    const variables = { slug: params.slug }
-    const {
-      data: { Post },
-    } = await client.query({ query, variables })
-    if (Post.length <= 0) {
-      // TODO: custom 404 error page with translations
-      const message = 'This post could not be found'
-      return error({ statusCode: 404, message })
```


Removing `asyncData` will also remove the feature of sending a 404 if there is no post!

Comment (Tirokk) [reply webapp/pages/post/_id/_slug/index.vue:140]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-16T11:05:00Z" title="Friday, August 16th 2019, 1:05:00 pm +02:00">Aug 16, 2019</time>_

---


I don't think that is true
Screenshot at 2019-08-16 13:04:24

**Review by Mogge — CHANGES_REQUESTED** [review]
Great @mattwr18! However I think it's quite important to send 404 if there is not Post available.. :cry:

**Review by Mogge — APPROVED** [review]
@mattwr18 you're right!
You're right

**Pull request closed without merging**